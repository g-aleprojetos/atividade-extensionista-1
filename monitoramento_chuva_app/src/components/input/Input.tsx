import React, {HTMLInputTypeAttribute, MutableRefObject} from 'react';
import * as S from './Input.styles';
import {TextoXSmall} from 'components/texto';
import colors from 'resources/colors';

export type Props = S.PropsInput & {
  id: string;
  label?: string;
  corTexto?: string;
  tipo: HTMLInputTypeAttribute;
  handleValue?: string | ReadonlyArray<string> | number | undefined;
  handleRef?: MutableRefObject<HTMLInputElement | null>;
  handleOnChange?: (_item: string) => void;
};

export const Input = (props: Props) => {
  const {
    id,
    label,
    corTexto = colors.black,
    tipo,
    handleValue,
    handleRef,
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    handleOnChange = () => {},
  } = props;

  const handleInput = (e: string) => {
    handleOnChange(e);
  };

  return (
    <S.Container data-testid={'test_input_container'} margin={props.margin}>
      {Boolean(label) && (
        <S.Label htmlFor={id}>
          <TextoXSmall cor={corTexto}>{label}</TextoXSmall>
        </S.Label>
      )}
      <S.ContainerInput
        data-testid={'test_input_containerInput'}
        type={tipo}
        ref={handleRef}
        onChange={e => handleInput(e.target.value)}
        value={handleValue}
        {...props}
      />
    </S.Container>
  );
};
