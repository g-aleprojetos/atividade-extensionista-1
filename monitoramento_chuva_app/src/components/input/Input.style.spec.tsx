import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './Input.styles';

describe('Input.styles', () => {
     test('Container DEVE ser igual ao snapshot', () => {
        const componente = renderer.create(<S.Container />).toJSON();
        expect(componente).toMatchSnapshot()
    });

    test('Label DEVE ser igual ao snapshot', () => {
        const componente = renderer.create(<S.Label />).toJSON();
        expect(componente).toMatchSnapshot()
    });

    test('ContainerInput DEVE ser igual ao snapshot', () => {
        const componente = renderer.create(<S.ContainerInput />).toJSON();
        expect(componente).toMatchSnapshot()
    });
});
