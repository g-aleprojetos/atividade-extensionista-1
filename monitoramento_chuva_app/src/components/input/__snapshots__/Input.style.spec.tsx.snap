// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0.813rem;
}

<div
  className="c0"
/>
`;

exports[`Input.styles ContainerInput DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 23.563rem;
  height: 2.125rem;
  padding: 0.5rem;
  border: 1px solid #000000;
  border-radius: 0.313rem;
  font-size: 16px;
}

<input
  className="c0"
/>
`;

exports[`Input.styles Label DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 0.5rem;
}

<label
  className="c0"
/>
`;
